src/main/java/me/karakelley/http/handlers/Application.java

```diff
@@ -17,10 +17,11 @@ public Application() {
 
   public Application(PublicDirectory publicDirectory) {
     this.publicDirectory = publicDirectory;
-    router.route(HttpMethod.POST, new NewResourceHandler(publicDirectory));
-    router.route(HttpMethod.GET, new StaticFilesHandler(publicDirectory, new HtmlPresenter()));
-    router.route(HttpMethod.PUT, new UpdateResourceHandler(publicDirectory));
-    router.route(HttpMethod.DELETE, new DeleteResourceHandler(publicDirectory));
+    Handler staticFileHandler = new StaticFilesHandler(publicDirectory, new HtmlPresenter());
+    router.route(HttpMethod.GET, staticFileHandler);
+    router.route(HttpMethod.POST, staticFileHandler);
+    router.route(HttpMethod.PUT, staticFileHandler);
+    router.route(HttpMethod.DELETE, staticFileHandler);
```

Comment:
I'm of two minds on this. On one hand, the granularity here is nice: I like how the full spectrum of requests supported by your server is at one "level", so (for example) it's easy to see that your server does not support HEAD/OPTIONS requests, and (as another example) if some other user of your server only wants to support read actions, they can just route in a GET with the StaticFilesHandler and not add POST/PUT/DELETE routes. I could see an argument that this is better than if all the methods were supported within StaticFilesHandler such that someone who only wants read actions would have to recreate the GET portion of that handler without all the rest.

On the other hand, there are other ways someone could get that read-only functionality (some kind of auth filter preventing write requests), and it still feels a little strange to me that, from the client perspective, your generic filesystem-y routes seem to get wired in at the same level of granularity as your domain-specific routes (i.e. /redirectme, /parse). The method signature (i.e. the lack of a specific path argument) does help, but I keep feeling like I want some sort of API like "on GET /redirectme do this, on GET /parse do that, and on anything else, here's a staticFilesHandler you can use to potentially do something in the public directory you're serving". Does that make sense? I don't think I'm explaining it well. @aaronlahey do you have any thoughts on this?